sandbox/apparmor/apparmor.go

```diff
@@ -374,6 +381,39 @@ func tryAppArmorParserFeature(parser, rule string) bool {
 	return true
 }
 
+// UpdateHomedirsTunable sets the AppArmor HOMEDIRS tunable to the list of the
+// specified directories. This directly affects the value of the AppArmor
+// @{HOME} variable. See the "/etc/apparmor.d/tunables/home" file for more
+// information.
+func UpdateHomedirsTunable(homedirs []string) error {
+	homeTunableDir := filepath.Join(ConfDir, "tunables/home.d")
+	tunableFilePath := filepath.Join(homeTunableDir, "snapd")
+
+	// If the file is not there and `homedirs` is empty, do nothing; this is
+	// not just an optimisation, but a necessity in Ubuntu Core: the
+	// /etc/apparmor.d/ tree is read-only, and attempting to create the file
+	// would generate an error.
+	if len(homedirs) == 0 && !osutilFileExists(tunableFilePath) {
+		return nil
+	}
+
+	if err := osMkdirAll(homeTunableDir, 0755); err != nil {
+		return fmt.Errorf("cannot create AppArmor tunable directory: %v", err)
+	}
+
+	contents := &bytes.Buffer{}
+	fmt.Fprintln(contents, "# Generated by snapd -- DO NOT EDIT!")
+	if len(homedirs) > 0 {
+		contents.Write([]byte("@{HOMEDIRS}+="))
+		separator := ""
+		for _, dir := range homedirs {
+			fmt.Fprintf(contents, `%s"%s"`, separator, dir)
+			separator = " "
+		}
+	}
```

Comment:
We could use `osutil.EnsureFileState()` here (or EnsureDirState but that is probably not what we need here) - the advantage is that it also ensures that a file is not written if the content did not change (but maybe that does not matter here?)

Reply:
Thanks, unfortunately the entire config file generation is triggered when any option changes. There is currently no smart code in the coreconfig that figures this out so each part of the coreconfig code needs to do it by itself. One easy way is to use EnsureFileState, another one can be to get the old value, compare with the new value and only write when changed.

To see for yourself:
```
$ sudo snap set system homedirs=/home/users
$ ls -l /etc/apparmor.d/tunables/home.d/r.d/tunables/home.d/snapd 
total 12
-rw-r--r-- 1 root root  63 Jul 18 16:15 snapd
[wait 2min]
$ sudo snap set system system.timezone=Europe/Amsterdam
[totally unreleated option above]
$ ls -l /etc/apparmor.d/tunables/home.d/
total 12
-rw-r--r-- 1 root root  63 Jul 18 16:18 snapd
```